### DISCUSSION TITLE
Qt keymap failure

### DISCUSSION THREAD

**Discussion by ElDifinitivo** [qutebrowser/qutebrowser#7274] [Q&A]
I come across this issue every couple of weeks or so. There's no qutebrowser log and no crash and it seems to happen sporadically; the only cause I can maybe link it to is opening a new tab and quickly using any combination of keybindings. It's not a guarantee though, so I couldn't comfortably tie it to that personally.

The only log I've ever seen have been system logs similar to this:

```
WARNING: Qt: failed to retrieve the virtual modifiers names from XKB
WARNING: Qt: failed to retrieve the virtual modifier map from XKB
WARNING: failed to compile a keymap
```

When this happens, my keyboard is entirely unusable in my Xorg session. However, I can jump to another TTY as the keyboard works outside of the current Xorg session. From there I kill Xorg and can go back to the first TTY and startx again.

```
$ qutebrowser --version

qutebrowser v2.5.1
Git commit:
Backend: QtWebEngine 5.15.2, based on Chromium 87.0.4280.144
Qt: 5.15.4 (compiled 5.15.2)

CPython: 3.10.5
PyQt: 5.15.5

sip: 6.1.0.dev2104271705
colorama: 0.4.4
jinja2: 3.0.3
pygments: 2.12.0
yaml: 6.0
adblock: 0.5.2
PyQt5.QtWebEngineWidgets: yes
PyQt5.QtWebEngine: 5.15.5
PyQt5.QtWebKitWidgets: no
pdf.js: 2.12.313 (/usr/share/pdf.js/build/pdf.js)
sqlite: 3.38.5
QtNetwork SSL: OpenSSL 1.1.1n  15 Mar 2022

Style: QFusionStyle
Platform plugin: xcb
OpenGL: AMD, 4.6 (Compatibility Profile) Mesa 22.1.1
Platform: Linux-5.15.45_1-x86_64-with-glibc2.32, 64bit
Linux distribution: Void Linux (void)
Frozen: False
Imported from /usr/lib/python3.10/site-packages/qutebrowser
Using Python from /usr/bin/python3
Qt library executable path: /usr/lib/qt5/libexec, data path: /usr/share/qt5

Paths:
cache: /home/<user>/.cache/qutebrowser
config: /home/<user>/.config/qutebrowser
data: /home/<user>/.local/share/qutebrowser
runtime: /run/user/1000/qutebrowser
system data: /usr/share/qutebrowser

Autoconfig loaded: no
Config.py: /home/<user>/.config/qutebrowser/config.py has been loaded
Uptime: 0:00:00
```

**Comment by The-Compiler** [comment]
Converting this to a discussion as this doesn't seem like anything that would be caused by qutebrowser's code or could be fixable by qutebrowser. I've never seen it, but I'm assuming you'd see the same in any other Qt application too.

**Reply by ElDifinitivo** [reply]
Yes, a discussion would be better. I have yet to experience anything like this with another Qt app before, but granted qutebrowser is the only one I use with any sort of regularity other than nheko. I was hoping if anyone else has ever experienced this before with either qutebrowser or any other Qt software.
